.github/workflows/build_and_test_workflow.yml

```diff
@@ -95,18 +96,21 @@ jobs:
         run: yarn osd bootstrap || yarn osd bootstrap
 
       - name: Run linter
+        if: matrix.group == 2 && matrix.os == 'ubuntu-latest'
         id: linter
         run: yarn lint
 
       - name: Validate NOTICE file
+        if: matrix.group == 2 && matrix.os == 'ubuntu-latest'
         id: notice-validate
         run: yarn notice:validate
 
-      - name: Run unit tests with coverage
+      - name: Run unit tests group ${{ matrix.group }} with coverage
         id: unit-tests
-        run: yarn test:jest:ci:coverage
+        run: yarn test:jest:ci:coverage  --ci-group=${{ matrix.group }}
 
       - name: Run mocha tests with coverage
+        if: matrix.group == 2
         id: mocha-tests
         run: yarn test:mocha:coverage
```

Comment:
This probably isn't wasting any time, but can be removed altogether. We've already migrated all core tests from mocha, it's only still supported for plugins.

Reply:
I will remove in in an upcoming PR

Reply:
https://github.com/opensearch-project/OpenSearch-Dashboards/issues/4476